vendor/net_gearman/Net/Gearman/AtomWorker.php

```diff
@@ -0,0 +1,152 @@
+            $currentTime = time();
+            foreach ($this->conn as $server => $socket) {
+                $worked = false;
+
+                try {
+                    $worked = $this->doWork($socket);
+                } catch (Net_Gearman_Exception $e) {
+                    unset($this->conn[$server]);
+                    $this->retryConn[$server] = $currentTime;
+                }
+                if ($worked) {
+                    $lastJob = time();
+                    $sleep = false;
+                }
+            }
+            $idle = false;
+            if ($sleep && count($this->conn)) {
+                foreach ($this->conn as $socket) {
+                    Net_Gearman_Connection::send($socket, 'pre_sleep');
+                }
+                $read = $this->conn;
+                @socket_select($read, $write, $except, $this->socket_timeout);
+                $idle = (0 == count($read));
+            }
+            $retryChange = false;
+            foreach ($this->retryConn as $s => $lastTry) {
+                if (($lastTry + $retryTime) < $currentTime) {
+                    try {
+                        $conn = Net_Gearman_Connection::connect($s);
+                        $this->conn[$s] = $conn;
+                        $retryChange = true;
+                        unset($this->retryConn[$s]);
+                        Net_Gearman_Connection::send($conn, 'set_client_id', ['client_id' => $this->id]);
+                    } catch (Net_Gearman_Exception $e) {
+                        $this->retryConn[$s] = $currentTime;
+                    }
+                }
+            }
+            if (0 == count($this->conn)) {
+                // sleep to avoid wasted cpu cycles if no connections to block on using socket_select
+                sleep(1);
+            }
+            if (true === $retryChange) {
+                // broadcast all abilities to all servers
+                foreach ($this->abilities as $ability => $timeout) {
+                    $this->addAbility(
+                        $ability, $timeout, $this->initParams[$ability]
+                    );
+                }
+            }
+            if (true == call_user_func($monitor, $idle, $lastJob)) {
+                $working = false;
+            }
+
+            if ($this->maxJobCountReached()) {
+                $working = false;
+            }
+        }
+    }
+
+    protected function maxJobCountReached()
+    {
+        if ($this->maxJobCount <= 0) {
+            return false;
+        }
+
+        if ($this->jobsCompleted >= $this->maxJobCount) {
+            $this->log(sprintf('Max job count reached: %u jobs completed.', $this->jobsCompleted), sfLogger::INFO);
+
+            return true;
+        }
+
+        return false;
+    }
+
+    protected function incrementJobsCompleted()
+    {
+        ++$this->jobsCompleted;
+    }
+
+    protected function complete($handle, $job, array $result)
+    {
+        $this->incrementJobsCompleted();
```

Comment:
Is there an advantage to using a `$this->incrementJobsCompleted()` method here instead of just using `++$this->jobsCompleted` directly?

Reply:
Just personal pref... Similar to using getters and setters within an object, it provides a unified interface for performing this function. Granted, incrementing an int is pretty simple and probably not necessary!  😄